src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -96,6 +97,50 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history }: ViewPro
     }
   };
 
+  const [isOptionsOpen, setOptionsOpen] = useState(false);
+  const [useLegacy, setUseLegacy] = useState(!getNewDiscoverSetting(storage));
+  const DiscoverOptions = () => (
```

Comment:
Could consider to move `DiscoverOptions` out of `DiscoverCanvas` for code readability and  reusability

Reply:
good point. ideally, there is a shelf life of this. until we match performance and feature parity of the legacy and current Discover